### DISCUSSION TITLE
CS8926 on static abstract interface method requiring type parameter, can't fufil

### DISCUSSION THREAD

**Discussion by Xyncgas** [dotnet/csharplang#6348] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/35312807/183228915-edccd000-1205-4d1f-a5bf-5dacd2711def.png)
![image](https://user-images.githubusercontent.com/35312807/183228938-4dcb55ae-2531-4aa0-9886-42d9c575b16f.png)
![image](https://user-images.githubusercontent.com/35312807/183229026-c53bb0b4-97e5-4bc7-b944-3a132da7bfdf.png)

**Comment by CyrusNajmabadi** [comment, marked as answer]
What's the question?  This is not allowed currently and the error is letting you know :-)

**Reply by Xyncgas** [reply]
It's not allowed? ohh
My question was the error tells me I need to use a typed parameter so I tried but I couldn't make it work
Since you said it's not possible I would like to ask how come I can't use the static abstract property in the method there should be nothing preventing me to... ohh

**Reply by bernd5** [reply]
I think the reason is, that the concrete type of the implementing interface is not known  in an interface default implementation. It could be handled like virtual generic, but ist not so easy...

**Reply by CyrusNajmabadi** [reply]
>  so I tried but I couldn't make it work

Could you show where you tried it?  Your example does not show accessing off of a Type-Parameter.   Thanks!

**Reply by Xyncgas** [reply]
On the first picture it shows the error message, on the second picture I added the qualification and it still shows the error message, which leads me thinking I might be unaware of the correct way to do it which I am humbled to learn

**Reply by CyrusNajmabadi** [reply]
Your second picture is not calling the method off of a type parameter. :-)

**Reply by Xyncgas** [reply]
Oh so can I do this within the interface or do I have to take a type that implements the interface to qualify it

**Reply by 333fred** [reply]
The latter.
